openedx/openedx-platform
# fix: staff debug actions depended on legacy courseware URL

## Description

The Staff Debug Actions didn't work in the Learning MFE
because the underlying JS depended on the URL being
formatted as /courses/<course_key>/... in order to
parse out the course key. This worked in the legacy
experience, but breaks in the chromeless xblock view,
which is rendered under the URL /xblock/<usage_key>/...

The fix is to explicitly pass the course key into the
templated courseware HTML as a data attribute.

## Commits

I'll squash these together before merging, so I recommend reviewing using the full diff. But if you're interested:

1. `fix: ...` : The bugfix itself.
2. `refactor: ...`: I started refactoring the StaffDebug class to de-dupe the parameters, but I decided it wasn't worth it. You can ignore this.
3. `revert: ...`: Reverts refactor commit. You can ignore this.
4. `style: ...`: Breaks some long lines.
5. `squahsme: ...`: Fixes JS tests affected by first commit.

## Supporting information

https://openedx.atlassian.net/browse/TNL-7955

## Testing instructions

Reproducing the bug:
* In devstack: `make frontend-app-learning-up`
* In Legacy Experience, nagivate to a block in a course that you have staff access to.
* Hit "Staff Debug Info". Override your score. Confirm that there is no error message. In your browser's network log, note the URL that was POSTed to when you tried overriding the score.
* Do the same thing in New Experience. You should see a red error message. In your browser's networok log, note that the POSTed URL is missing the `/courses/<course_key>` part.

Confirming the fix:
* Checkout this branch
* Devstack: `make lms-restart lms-shell`
* Within the :LMS shell: `NO_PREREQ_INSTALL=1 paver update_assets lms`
* Get some coffee...
* Navigate to the New Experience and try to override your score again. There should be no error message, and the POSTed URL should the same as what you saw for the Legacy Experience.

## Deadline

CAT-3 ; sooner is better
